cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1185,8 +1189,16 @@ private void deleteApp(ApplicationId appId, ApplicationSpecification spec) throw
     deleteAppMetadata(appId, spec);
     store.deleteWorkflowStats(appId);
     deletePreferences(appId, spec);
+
+    store.markDeleteApplication(appId);
+    // TODO : remove this in CDAP-20333 to test end to end delete
     store.removeApplication(appId);
-    
+    try {
+      appDeletionPublisher.publishAppDeletionEvent(appId);
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("Failed to publish app deletion request for entity id %s", appId), e);
```

Comment:
Why this rethrow and why plain RuntimeException? This would convert everything into 500.

Reply:
so I think this exception is thrown at L83 in AppDeletionPublisher https://github.com/cdapio/cdap/pull/14897/commits/03e100fad30cf2fab3d9b689be797c8449b577db#diff-39cb51e00d3dda007ea7cfca088c48dc7fad2095711ade8daf4fdd55ae3fc213R81-R83 - we can eliminate this try - catch block. My point is to make sure we do not proceed ahead if the publish app deletion message fails

Reply:
If the  original exception will be thrown out it should stop processing, you should not need to wrap exception here, at least I don't see the reason